src/types/fieldExtractor.ts

```diff
@@ -81,3 +134,43 @@ export const resolveEthereumTrackIdOverrides = (nft: NFT, contract: NftFactory):
   }
   return ethereumTrackId(nft.contractAddress, trackId);
 }
+
+export const resolveArtworkUrlOverrides = (nft: NFT, contract: NftFactory): string => {
+  const override = contract.typeMetadata?.overrides?.extractor?.artworkUrl;
+
+  if (override === ArtworkUrlExtractorTypes.USE_ARTWORK_URL_OVERRIDE) {
+    const track = contract.typeMetadata?.overrides?.track;
+    const url = (track as any).lossyArtworkURL;
+    if (!url) {
+      throw new Error('track lossy artwork url override not provided');
+    }
+    return url;
+  }
+  return nft.metadata.image
+}
+
+export const resolveArtistNameOverrides = (nft: NFT, contract: NftFactory): string => {
+  const artistNameExtractorOverride = contract.typeMetadata?.overrides?.extractor?.artistName;
+  if (!artistNameExtractorOverride || (artistNameExtractorOverride !== ArtistNameExtractorTypes.ATTRIBUTES_TRAIT_MUSICIAN)) {
+    return contract.platformId;
+  }
+  return artistNameExtractor(contract)(nft);
+}
+
+export const resolveArtistIdOverrides = (nft: NFT, contract: NftFactory): string => {
+  const artistIdExtractorOverride = contract.typeMetadata?.overrides?.extractor?.artistId;
+  if (artistIdExtractorOverride) {
+    if (artistIdExtractorOverride === ArtistIdExtractorTypes.USE_PLATFORM_ID) {
+      return contract.platformId;
+    }
+    else if (artistIdExtractorOverride === ArtistIdExtractorTypes.USE_PLATFORM_AND_ARTIST_NAME) {
+      return `${contract.platformId}/${slugify(resolveArtistNameOverrides(nft, contract))}`;
+    }
+  }
+
+  const artistIdOverride = contract.typeMetadata?.overrides?.artist?.artistId;
+  if (!artistIdOverride) {
+    throw new Error('must directly provided an artist ID override, or an artist ID extractor override');
```

Comment:
should we default to platformId as before? or otherwise i think this may break chaos so we may need to add an artist id override on chaos too

Reply:
re chaos - nvm, i see it doesn't actually use multi-track-multiprint, i thought it did